hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/entity/AccountBuilderImpl.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.token.impl.entity;
+
+import static com.hedera.node.app.service.token.entity.Account.HBARS_TO_TINYBARS;
+
+import com.hedera.node.app.service.token.entity.Account;
+import com.hedera.node.app.service.token.entity.AccountBuilder;
+import com.hedera.node.app.spi.key.HederaKey;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * An implementation of {@link AccountBuilder} for building Account instances. This class is
+ * <strong>not</strong> exported from the module. FUTURE: Should be moved to token-service-impl
+ * module
+ */
+public class AccountBuilderImpl implements AccountBuilder {
+    private final Account copyOf;
+
+    // These fields are the ones that can be set in the builder
+    // FUTURE: Replace the empty KeyList we use for 0.0.800 and hollow
+    // accounts with an empty Optional
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    private Optional<HederaKey> key;
+
+    private long expiry;
+    private long balance;
+    private String memo;
+    private boolean deleted;
+    private boolean receiverSigRequired;
+    private long numberOfOwnedNfts;
+    private int maxAutoAssociations;
+    private int usedAutoAssociations;
+    private int numAssociations;
+    private int numPositiveBalances;
+    private long ethereumNonce;
+    private long stakedToMe;
+    private long stakePeriodStart;
+    private long stakedNum;
+    private boolean declineReward;
+    private long stakeAtStartOfLastRewardedPeriod;
+    private long autoRenewAccountNumber;
+    private long autoRenewSecs;
+
+    /**
+     * Create a builder for creating {@link Account}s, using the given copy as the basis for all
+     * settings that are not overridden.
+     *
+     * @param copyOf The instance to copy
+     */
+    public AccountBuilderImpl(@NonNull Account copyOf) {
```

Comment:
Does this mean, we always need an `Account`, if we want to create an `AccountBuilderImpl`? How is the first `Account` created? Is there no need to create an `Account` from scratch?